lib/ht_traject/print_holdings_api.rb

```diff
@@ -1,15 +1,18 @@
 require_relative "../services"
+require "httpx/adapters/faraday"
 
 module HathiTrust
   class PrintHoldingsAPI
 
     def self.connection 
-      @conn ||= Faraday.new( url: ENV["HOLDINGS_API_URL"] ) do |builder|
+      Thread.current[:holdings_api_conn] ||= Faraday.new( url: ENV["HOLDINGS_API_URL"] ) do |builder|
         builder.request :json
-        builder.response :json
         # TODO: Perhaps if it fails we should retry instead of immediately raising an
         # error?
         builder.response :raise_error
+        builder.adapter :httpx
+      end.tap do |conn|
+        Services.logger.info("Allocated new connection #{conn} for print holdings api")
```

Comment:
@moseshll I'm inclined to keep this logging & the debug logging below (although it is quite verbose when using `--verbose`), but let me know if you think I should remove it.

Reply:
Yes, keep it.